vowpalwabbit/learner.h

```diff
@@ -690,7 +700,32 @@ struct reduction_learner_builder
     return *this;
   }
 
-  learner<DataT, ExampleT>* build() { return this->_learner; }
+  learner<DataT, ExampleT>* build(VW::io::logger* logger = nullptr)
+  {
+    if (logger != nullptr)
+    {
+      prediction_type_t& in_pred_type = this->_learner->_input_pred_type;
+      prediction_type_t& base_out_pred_type = this->_learner->learn_fd.base->_output_pred_type;
+      label_type_t& out_label_type = this->_learner->_output_label_type;
+      label_type_t& base_in_label_type = this->_learner->learn_fd.base->_input_label_type;
+      if (in_pred_type != base_out_pred_type)
+      {
+        logger->err_warn(
+            fmt::format("Input prediction type: {} of reduction: {} does not match output prediction type: {} of base "
+                        "reduction: {}.",
+                to_string(in_pred_type).data(), this->_learner->name, to_string(base_out_pred_type).data(),
```

Comment:
`.data()` should not be needed. You may need this [include](https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/vowpalwabbit/vw_string_view_fmt.h)

```suggestion
                to_string(in_pred_type), this->_learner->name, to_string(base_out_pred_type),
```

Reply:
Think that can just be removed